### PR TITLE
Feat/spt-77/add usermenu on header

### PR DESCRIPTION
**Pull request by tebogo-bbd**
user logged out
usermenu logged out v2

user logged in 
usermenu logged in v3

### CODE REVIEW

File: frontend/components/header.js [hunk 1]
```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
   connectedCallback() {
     const rawUsername = localStorage.getItem('signedIn') === "true" ? localStorage.getItem('username') : "there";
     const username = escapeHTML(rawUsername);
-
+    const profilePicture = username === 'there' ? `assets/images/default-avatar.jpeg` : `https://github.com/${username}.png`;
```
Comment (sudosf) [inline frontend/components/header.js]: May we plz create a new variable called `const isSignIn` and use that instead of checking twice

File: frontend/components/header.js [hunk 2]
```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
...
+    const links = localStorage.getItem('signedIn') === "true" 
+                ? ` <a href="leaderboard.html">leaderboard</a>
```
Comment (sudosf) [inline frontend/components/header.js:5]: Just a UI suggestion, could we use icons for the other links as darkmode has an icon. for the sake of consistency in the menu. We can do it in another PR though if not possible in this one

File: frontend/components/header.js [hunk 3]
```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
...
+            <header>
+                <img class="logo" src=${Icons.logo} alt="logo" />
+                <div class="dropdown">
+                    <img src=${profilePicture} alt="Avatar" class="avatar">
+                    <div class="dropdown-content">
+                        <h6 class="dropdown-name">Hello, ${username}</h6>
+                        <div class="dropdown-links">
+                            <a href="#" id="theme-toggle-btn"> 
+                                    <img class="moon" src=${Icons.moon} alt="theme-toggler" />
+                                    <span>dark mode</span>
```
Comment (sudosf) [inline frontend/components/header.js:20]: Span is not really a semantic element, we might be questioned why we used it. i would suggest using a <p> tag

File: frontend/components/header.js [hunk 4]
```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
...
+                                </a>
+                            ${links}
+                        </div>
+                    </div>
+                </div>
+            </header>
         `;
-    }
+
+    this.querySelector('h6').addEventListener('click', this.toggleDropdown);
```
Comment (sudosf) [inline frontend/components/header.js]: May we plz add the event listener to the parent element (i.e. the dropdown container)

File: frontend/css/style.css [hunk 5]
```diff
@@ -272,3 +272,85 @@ progress::-webkit-progress-bar {
     height: 100%;
     border-radius: 1.5em;
 }
+
+/** Subjects on the homepage **/
```
Comment (sudosf) [inline frontend/css/style.css:244]: subject container styles are duplicate, may we plz remove them from line 276 to 307

**Review by sudosf — CHANGES_REQUESTED** [review]
left a few comments

File: frontend/components/header.js [hunk 6]
```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
...
+   
+  }
+  toggleDropdown = () => {
+    const dropdownMenu = this.querySelector('.dropdown-menu');
+    dropdownMenu.style.display = dropdownMenu.style.display === 'block' ? 'none' : 'block';
```
Comment (sudosf) [inline frontend/components/header.js]: May we plz structure this tenary so it's more readable
```suggestion
    dropdownMenu.style.display = dropdownMenu.style.display ===  'block' 
                                                      ? 'none' 
                                                      : 'block';
```

File: frontend/components/header.js (hunk 2 above)
Comment (tebogo-bbd) [reply frontend/components/header.js:5]: let's leave it for another PR

File: frontend/components/header.js (hunk 3 above)
Comment (tebogo-bbd) [reply frontend/components/header.js:20]: I think span is justified here, basically it's there for flex to work properly

File: frontend/components/header.js (hunk 4 above)
Comment (tebogo-bbd) [reply frontend/components/header.js]: oh yha that's not even used no more will wack it out

File: frontend/components/header.js (hunk 6 above)
Comment (tebogo-bbd) [reply frontend/components/header.js]: gonna remove this part of code as it's no longer used

Comment (tebogo-bbd) [reply frontend/components/header.js]: and that is readable

File: frontend/components/header.js (hunk 2 above)
Comment (tebogo-bbd) [reply frontend/components/header.js:5]: I'll do it after the conversion

File: frontend/css/style.css (hunk 5 above)
Comment (tebogo-bbd) [reply frontend/css/style.css:244]: I am gonna remove the styles you nested inside .main

**Review by Vutlhari-Mashimbyi — APPROVED** [review]

**Comment by Vutlhari-Mashimbyi** [comment]
Great
